pkg/asset/cluster/cluster.go

```diff
@@ -102,3 +102,9 @@ func (c *Cluster) Generate(parents asset.Parents) error {
 func (c *Cluster) Files() []*asset.File {
 	return []*asset.File{c.file}
 }
+
+// SetState sets the appropriate fields from given unmarshalled files
+func (c *Cluster) SetState(files []*asset.File) error {
```

Comment:
This approach breaks all the code where we were using the asset to get information not reading the file like this https://github.com/openshift/installer/blob/master/pkg/asset/manifests/operators.go#L175

Reply:
We can only marshal/unmarshal WritableAssets. If there is a non-writable-asset dependency, then 'generate' will be called for the 'dirty' tree branch anyway.

Reply:
When loading installconfig asset you only load the files. but the code https://github.com/openshift/installer/blob/master/pkg/asset/manifests/operators.go#L175 uses the embedded `type.installconfig` directly not the file (which we want to continue to support).

This new SetState doesn't allow that.

Reply:
https://github.com/openshift/installer/pull/388/files#diff-74e3d89f2a13e465091f905f65cfc646R144 doesn't seem to be doing what you are saying...

Reply:
Yes, you are right. Its still a TODO :). Sorry if that confused you.. I should have put in a comment there.